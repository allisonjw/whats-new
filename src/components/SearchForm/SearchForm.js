import React, {Component} from 'react';
import './SearchForm.scss';

class SearchForm extends Component {
    constructor() {
        super()
        this.state = {
            input: ""
        }
    }

handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
}

handleClick = e => {
    e.preventDefault();
    const searchText = this.state.input.toLowerCase();
    this.props.displayNewsSearch(searchText);
    this.setState({ input: "" });
}

  render() {
    return (
        <form>
            <input 
              type="text" 
              placeholder="Enter a news topic" 
              name="input"
              value={this.state.input}
              onChange={e => this.handleChange(e)}
            />
            <button 
              type="submit"
              name="submit"
              onClick={e => this.handleClick(e)}
            >Search Now
            </button>
        </form>
    )
  }
}


export default SearchForm;