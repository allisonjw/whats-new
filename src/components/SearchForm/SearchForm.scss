form {
  display: flex;
  justify-content: center
}

%formElementStyles {
  border: 2px solid black;
  font-size: 1.2em;
  height: 35px;
  margin: 5px;
  width: 21%;
}

input {
  @extend %formElementStyles;
}

button {
  @extend %formElementStyles;
  border-radius: 16px;
}
  