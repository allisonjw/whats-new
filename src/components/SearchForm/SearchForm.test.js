import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';
import { GiStarFormation } from 'react-icons/gi';

describe('SearchForm', () => {
  let wrapper;
  let mockDisplayNewsSearch = jest.fn();

  beforeEach(() => {
    mockDisplayNewsSearch = jest.fn();
    wrapper = shallow(<SearchForm displayNewsSearch={mockDisplayNewsSearch} />)
  });

  // console.log(wrapper.debug());
  // remember to use this later
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: { name: 'input', value: 'NASA'} };
    const expected = 'NASA';
  
    wrapper.instance().handleChange(mockEvent);
  
    expect(wrapper.state('input')).toEqual(expected);
  });

  it('should reset state after form is cleared', () => {
    const startState = {input: 'NASA'};
    const expected = { input: '' };

    wrapper.setState(startState);
    expect(wrapper.instance().resetInput);
    wrapper.state(expected);
  });

  it('should run handleClick when the button is clicked', () => {
    wrapper.instance().handleClick = jest.fn();
    wrapper.instance().forceUpdate();
    const mockEvent = { preventDefault: jest.fn() };
  
    wrapper.find('button').simulate('click', mockEvent);
  
    expect(wrapper.instance().handleClick).toHaveBeenCalledWith(mockEvent);
  });

});