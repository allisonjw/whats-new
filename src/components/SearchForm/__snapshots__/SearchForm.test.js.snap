// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    name="input"
    onChange={[Function]}
    placeholder="Enter a news topic"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Search Now
  </button>
</form>
`;
